src/caches/beliefpropagationcache.jl

```diff
@@ -106,3 +106,21 @@ function region_scalar(bp_cache::BeliefPropagationCache, pe::PartitionEdge)
   sequence = contraction_sequence(ts; alg="optimal")
   return contract(ts; sequence)[]
 end
+
+function rescale_messages(bp_cache::BeliefPropagationCache, pes::Vector{<:PartitionEdge})
+  bp_cache = copy(bp_cache)
+  mts = messages(bp_cache)
+  for pe in pes
+    me, mer = only(mts[pe]), only(mts[reverse(pe)])
```

Comment:
Maybe I've lost track but when you call `only(mts[pe]), only(mts[reverse(pe)])`, is that assuming something about the messages, such as that the messages are just a single ITensor?

Reply:
Yeah it's assuming they are a single ITensor but stored as a `Vector{<:ITensor}` (which is the typical storage for for the `BeliefPropagationCache` )

Reply:
Thanks for the reminder. It seems like we should abstract that away to some extent, for example maybe we could define `message_overlap(bp_cache, edge::PartitionEdge)`. That would hide the detail of how `Vector{<:ITensor}` is handled, and then also generalize to things like MPS/TNS messages.

Reply:
Gotcha, I have now done it that way